core/meta/src/TClass.cxx

```diff
@@ -1399,10 +1399,12 @@ void TClass::Init(const char *name, Version_t cversion,
    }
 
    if (givenInfo) {
-      if (!gInterpreter->ClassInfo_IsValid(givenInfo) ||
-          !(gInterpreter->ClassInfo_Property(givenInfo) & (kIsClass | kIsStruct | kIsNamespace)) ||
-          (!gInterpreter->ClassInfo_IsLoaded(givenInfo) && (gInterpreter->ClassInfo_Property(givenInfo) & (kIsNamespace))) )
-      {
+      R__LOCKGUARD(gInterpreterMutex);
```

Comment:
This is clearer .. albeit in practice this is a no-op as Init is always called with the lock already taken (Init should be called only from the constructor and they should all be taking the lock before hand).